TypeScript.YAML-tmTheme

```diff
@@ -7,7 +7,7 @@ uuid: 91489F9C-F403-4CF0-993D-EAAF9149E40E
 settings:
 - scope: storage.modifier, storage.type, keyword.control, keyword.other, keyword.operator.expression, keyword.operator.new, keyword.generator.asterisk, punctuation.definition.template-expression
   settings: { vsclassificationtype: keyword }
-- scope: support.type, constant.language, variable.language
+- scope: support.type, constant.language, variable.language, variable.language variable.parameter

```

Comment:
This needs to be
`variable.parameter variable.language` instead ? since the scope of this in parameter position is:
```
Scope:                         source.ts
                               meta.function.ts
                               meta.parameters.ts
                               variable.parameter.ts
                               variable.language.this.ts
------
```

Reply:
It doesn't work when flipped. As far as I understand, the two cases that occur are:
1) the scope is:
```
Scope:                         source.ts
                               meta.function.ts
                               meta.parameters.ts
                               variable.parameter.ts
                               variable.language.this.ts
```
In this case we start with the last scope, `variable.language.this.ts`, which matches the `variable.language` rule and we get the right classification.
2) the scope is:
```
Scope:                         source.ts
                               meta.function.ts
                               meta.parameters.ts
                               variable.language.this.ts
                               variable.parameter.ts
```
Here `variable.parameter.ts` (last scope) matches `variable` but also `variable.language.this.ts variable.parameter.ts` (last 2 scopes) matches `variable.language variable.parameter` which is a more specific rule. So the second one is chosen.